tests/HHClientLinterTestTrait.hack

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use namespace HH\Lib\{Regex, Str};
+
+trait HHClientLinterTestTrait {
+  use LinterTestTrait;
+
+  abstract const HHClientLinter::TConfig CONFIG;
+
+  /**
+   * The temporary directory to include testing source files to lint.
+   *
+   * Note that the temporary directory must be under the current directory,
+   * otherwise hh_client will not work.
+   */
+  private static function temporarySourceDirectory():string {
+    return __DIR__.'/../.var/tmp/hhast/'.(Regex\first_match(static::class, re"/\\\\(\w+)$/")[1] ?? static::class);
```

Comment:
Regex seems a bit overkill and hard to read - `C\lastx(Str\split(static::class, "\\"))` is the more common pattern (and works both with and without namespaces)